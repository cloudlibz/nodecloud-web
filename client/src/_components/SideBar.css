.ui.sidebar.vertical.left.menu {
  position: fixed;
  left: 0;
  top: 65px;
  height: calc(100vh - 65px) !important;
  box-shadow: none;
  transition: all 300ms ease-in-out;
}
.ui.sidebar.vertical.left.menu.navActive {
  left: 0;
}
