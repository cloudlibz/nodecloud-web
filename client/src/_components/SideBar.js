import React, { Component } from 'react';
import './SideBar.css';
import { history } from '../_helpers/history.js';

class SideBar extends Component {
  constructor(props) {
    super(props);
    this.state = { visible: true };
    this.navActiveState = true;
    this.changeSelectedDashboardService = this.changeSelectedDashboardService.bind(
      this,
    );
  }

  changeSelectedDashboardService(serviceName) {
    history.push(`/${serviceName}`);
    this.props.changeSelectedDashboardService(serviceName);
  }

  render() {
    return (
      <div
        className={
          this.state.visible
            ? 'ui sidebar vertical left menu overlay visible navActive'
            : 'ui sidebar vertical left menu overlay visible'
        }
        id="site-sideBar"
      >
        <div className="ui accordion">
          <a href="#!" className="item" onClick={() => history.push('/home')}>
            <b>Home</b>
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('virtualmachines')}
          >
            Virtual Machines
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('virtualnetworks')}
          >
            Virtual Networks
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('database')}
          >
            Database
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('storage')}
          >
            Storage
          </a>
          <a
            href="#!"
            className="item"
            onClick={() => this.changeSelectedDashboardService('security')}
          >
            Security
          </a>
        </div>
      </div>
    );
  }
}

export default SideBar;
