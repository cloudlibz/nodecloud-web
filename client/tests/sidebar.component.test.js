import TestRenderer from 'react-test-renderer';
import SideBar from '../src/_components/SideBar'
import React from 'react'


describe("<SideBar /> Unit Tests", () => {

    it("renders correctly", () => {
        var testFunction = jest.fn();
        var component = TestRenderer.create( <
            SideBar changeSelectedDashboardService = {
                testFunction
            }
            />
        );
        expect(component).toMatchSnapshot();
    });
});