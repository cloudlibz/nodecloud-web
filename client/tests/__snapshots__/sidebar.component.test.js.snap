// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> Unit Tests renders correctly 1`] = `
<div
  class="ui sidebar vertical left menu overlay visible navActive"
  id="site-sideBar"
>
  <div
    class="ui accordion"
  >
    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      <b>
         Home 
      </b>

    </a>

    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      Virtual Machines 
    </a>

    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      Virtual Networks 
    </a>

    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      Database 
    </a>

    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      Storage 
    </a>

    <a
      class="item"
      href="#!"
      onClick={[Function]}
    >
      Security 
    </a>

  </div>

</div>
`;